microsoftgraph/microsoft-graph-explorer-v4
# Add headers to snippet request

## Overview
Fixes #1657 

### Demo

Optional. Screenshots, `curl` examples, etc.

### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

- It is incorrect to send content-type: application/json for GET requests, so we skip adding this content-type if the request method is GET. 

- We are sending over content-type: application/json by default for all requests as we currently only support JSON request body.
  If we start supporting other content-types in future, we won't need to change the way we handle content-types as that is 
  covered by this PR.

- The responses for Javascript snippets expected to have headers, do not have these headers except for GET queries. 
  This query for instance: 
      **GET https://graph.microsoft.com/v1.0/applications?$search="displayName:Web"&$count=true
      ConsistencyLevel: eventual**
   will return the correct Javascript snippet with the header included. 
   
   However, Javascript snippet response for queries with methods other than GET, do not include headers.
   For instance:
         **POST https://graph.microsoft.com/v1.0/users
         Content-type: application/json
         DummyHeader: dummyValue**
         (https://developer.microsoft.com/en-us/graph/graph-explorer?request=users&method=POST&version=v1.0&GraphUrl=https://graph.microsoft.com&requestBody=eyJhY2NvdW50RW5hYmxlZCI6dHJ1ZSwiY2l0eSI6IlNlYXR0bGUiLCJjb3VudHJ5IjoiVW5pdGVkIFN0YXRlcyIsImRlcGFydG1lbnQiOiJTYWxlcyAmIE1hcmtldGluZyIsImRpc3BsYXlOYW1lIjoiTWVsaXNzYSBEYXJyb3ciLCJnaXZlbk5hbWUiOiJNZWxpc3NhIiwiam9iVGl0bGUiOiJNYXJrZXRpbmcgRGlyZWN0b3IiLCJtYWlsTmlja25hbWUiOiJNZWxpc3NhRCIsInBhc3N3b3JkUG9saWNpZXMiOiJEaXNhYmxlUGFzc3dvcmRFeHBpcmF0aW9uIiwicGFzc3dvcmRQcm9maWxlIjp7InBhc3N3b3JkIjoiMTJmNTU1ODUtNjJjYy0zZThkLWE0NmMtOTM4ZDA4OWI1YTJkIiwiZm9yY2VDaGFuZ2VQYXNzd29yZE5leHRTaWduSW4iOmZhbHNlfSwib2ZmaWNlTG9jYXRpb24iOiIxMzEvMTEwNSIsInBvc3RhbENvZGUiOiI5ODA1MiIsInByZWZlcnJlZExhbmd1YWdlIjoiZW4tVVMiLCJzdGF0ZSI6IldBIiwic3RyZWV0QWRkcmVzcyI6IjkyNTYgVG93bmUgQ2VudGVyIERyLiwgU3VpdGUgNDAwIiwic3VybmFtZSI6IkRhcnJvdyIsIm1vYmlsZVBob25lIjoiKzEgMjA2IDU1NSAwMTEwIiwidXNhZ2VMb2NhdGlvbiI6IlVTIiwidXNlclByaW5jaXBhbE5hbWUiOiJNZWxpc3NhREBjb250b3NvLmNvbSJ9&headers=W3sibmFtZSI6IkR1bW15SGVhZGVyIiwidmFsdWUiOiJkdW1teVZhbHVlIn0seyJuYW1lIjoiQ29udGVudC10eXBlIiwidmFsdWUiOiJhcHBsaWNhdGlvbi9qc29uIn1d)
   The response for all other snippets have the dummy header:
   
![image](https://user-images.githubusercontent.com/45680252/172383434-009d2473-d503-40ce-a3b7-d61dd21080f9.png)
    The Javascript snippet does not have this header:
    
![image](https://user-images.githubusercontent.com/45680252/172383740-2b95b34e-9d3d-405c-beb8-543a12c80666.png)





## Testing Instructions

* How to test this PR
Check out this branch and run it locally
Add request headers on the request headers tab
Click on code snippets tab
Notice that the snippet response has all headers
